File: app/src/main/java/com/xxp/miaoview/di/component/MainComponent.java
package com.xxp.miaoview.di.component;

import dagger.Component;

import com.jess.arms.di.component.AppComponent;

import com.xxp.miaoview.di.module.MainModule;

import com.jess.arms.di.scope.ActivityScope;
import com.xxp.miaoview.mvp.ui.activity.MainActivity;

@ActivityScope
@Component(modules = MainModule.class, dependencies = AppComponent.class)
public interface MainComponent {
    void inject(MainActivity activity);
}

File: Mine/src/main/java/com/xxp/mine/di/component/MainComponent.kt
package com.xxp.mine.di.component;

import dagger.Component;
import com.jess.arms.di.component.AppComponent;

import com.xxp.mine.di.module.MainModule;

import com.jess.arms.di.scope.ActivityScope;
import com.xxp.mine.mvp.ui.activity.MainActivity;

@ActivityScope
@Component(modules = arrayOf(MainModule::class), dependencies = arrayOf(AppComponent::class))
interface MainComponent {
    fun inject(activity: MainActivity)
}
